### ISSUE TITLE
Add Transaction Semantics

### ISSUE THREAD

**Issue by jtaubensee** [Azure/azure-service-bus-dotnet#6]
_From @vinaysurya on October 12, 2016 2:12_

_Copied from original issue: Azure/azure-messaging-dotnet#10_

**Comment by jtaubensee** [comment]
.NET Core is now able to support transactions, so we will need to provide transaction support using AMQP in our service.

**Comment by SeanFeldman** [comment]
So [2.0](https://github.com/dotnet/corefx/issues/2949)

**Comment by affandar** [comment]
+1

Transaction support is necessary for workflow scenarios e.g. durabletasks and WF. Right now we are blocked on porting durabletasks to dotnet core due to this issue.

**Comment by jtaubensee** [comment]
Yeah so while transactions will be coming to .NET Core soon, we will still need to add support for transactions using AMQP in our service.

Hence, the need for this section in the article here: https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-amqp-dotnet#unsupported-features-restrictions-and-behavioral-differences

Based on the current priorities in this library (shipping our first preview), I expect that this will take a little while for us to complete.

**Comment by jtaubensee** [comment]
Created a service level issue at [azure-service-bus/#2](https://github.com/Azure/azure-service-bus/issues/2) to track.

**Comment by ElanHasson** [comment]
System.Transaction is now available. [corefx/#2949](https://github.com/dotnet/corefx/issues/2949)

**Comment by SeanFeldman** [comment]
@ChristianWolf42 this is the only issue assigned to [`3.0.0` milestone](https://github.com/Azure/azure-service-bus-dotnet/milestone/9). Is there a tentative timeline? Thank you.

**Comment by ChristianWolf42** [comment]
We did quite some work to enable transactions in November and hope to wrap it up and deliver the feature early next year. Hopefully first quarter of 2018.

**Comment by nemakam** [comment]
The PR is raised. Waiting for service side changes to be out.
#421 

@SeanFeldman 
>A few questions:
>- Will this address the issue of 100 messages limit?

No. That is service side limit that is not going to change with this.

>- Will this include send-via support?

No. In the first set of implementation, Send-via hasn't been added.

**Comment by SeanFeldman** [comment]
👍 for the first one. 😿 for the second @nemakam 
Is there a separate issue for Send-Via or should I raise it?

**Comment by SeanFeldman** [comment]
Also, should `OnHold` label be removed? 🙂

**Comment by SeanFeldman** [comment]
Raised an issue @nemakam (#423) as there was none.
